test/unit/update.js

```diff
@@ -679,6 +727,34 @@ exports['Tessel.update'] = {
     });
   },
 
+  configurationShouldNotBeSaved: function(test) {
+    var updatePath = `/tmp/${updates.OPENWRT_BINARY_FILE}`;
+
+    this.exec = this.sandbox.stub(this.tessel.connection, 'exec', (command, handler) => {
+      handler(null, this.tessel._rps);
+      setImmediate(() => {
+        this.tessel._rps.stdout.emit('data', new Buffer('Upgrade completed'));
+        this.tessel._rps.emit('close');
+      });
+    });
+
+    this.openStdinToFile = this.sandbox.stub(commands, 'openStdinToFile');
+    this.sysupgradeNoSaveConfig = this.sandbox.stub(commands, 'sysupgradeNoSaveConfig');
+
+    this.tessel.updateOpenWRT({
+      n: true
+    }, this.newImage.openwrt).then(() => {
+      test.equal(this.openStdinToFile.callCount, 1);
+      test.equal(this.sysupgradeNoSaveConfig.callCount, 1);
+      test.equal(this.openStdinToFile.lastCall.args[0], updatePath);
+      test.equal(this.sysupgradeNoSaveConfig.lastCall.args[0], updatePath);
+      test.done();
```

Comment:
Similar to the test I described above, add a test that asserts the right args were used